service/iot-service-client/src/main/java/com/microsoft/azure/sdk/iot/service/transport/amqps/CbsSenderLinkHandler.java

```diff
@@ -115,6 +115,7 @@ public int sendAuthenticationMessage(UUID authenticationMessageCorrelationId)
         {
             TokenRequestContext context = new TokenRequestContext().addScopes(IOTHUB_PUBLIC_SCOPE);
             this.currentAccessToken = credential.getToken(context).block();
+            //noinspection ConstantConditions
```

Comment:
@timtay-microsoft Here is your reminder about this method.

Reply:
Okay, so line 117 may return a null value here, so we would be better off checking if this.currentAccessToken is null and logging an error and returning -1 from this method

Reply:
```credential``` is a user provided ```TokenCredential``` interface implementation, so a user could hypothetically pass in an implementation of the interface that just always returns null